src/Tools/dotnet-monitor/Egress/EgressService.cs

```diff
@@ -84,23 +94,34 @@ public async Task<EgressResult> EgressAsync(string providerName, Func<Stream, Ca
             return new EgressResult(value);
         }
 
-        private IEgressProviderInternal GetProvider(string providerName)
+        private string GetProviderType(string providerName)
         {
-            if (!_providerTypeMap.TryGetValue(providerName, out string providerType))
+            if (!_providerNameToTypeMap.TryGetValue(providerName, out string providerType))
             {
                 throw new EgressException(string.Format(CultureInfo.CurrentCulture, Strings.ErrorMessage_EgressProviderDoesNotExist, providerName));
             }
+            return providerType;
+        }
 
-            if (!_optionsTypeMap.TryGetValue(providerType, out Type optionsType))
+        private IEgressProviderConfigurationProvider GetConfigProvider(string providerType)
```

Comment:
I think this method and `GetProvider` can be merged into one again since they are not used separately.